include/cucumber-cpp/internal/step/BreakpointSupport.hpp

```diff
@@ -0,0 +1,40 @@
+#ifndef CUKE_BREAKPOINTSUPPORT_HPP_
+#define CUKE_BREAKPOINTSUPPORT_HPP_
+
+#include "../CukeDll.hpp"
+
+namespace cucumber {
+namespace internal {
+
+#if defined(_MSC_VER)
+// The ", true" makes the expression to be of type bool, which is required
+// for the CUKE_BREAK_INTO_DEBUGGER macro
+#define CUKE_BREAK_INTO_DEBUGGER (__debugbreak(), true)
+
+#elif defined(__clang__) && defined(__APPLE__) && !TARGET_OS_IPHONE
+#define CUKE_BREAK_INTO_DEBUGGER                                     \
+  _Pragma("clang diagnostic push")                                   \
+  _Pragma("clang diagnostic ignored \"-Wgnu-statement-expression\"") \
+    ({ asm("int3"); true; })                                         \
+  _Pragma("clang diagnostic pop")
+
+#else // Generic Unix
+#include <signal.h>
+#include <unistd.h>
```

Comment:
It's also probably not such a good idea to `#include` from within
```c++
namespace cucumber {
namespace internal {
```
This will cause problems for everybody including these headers later. You should `#include` this at the top-level namespace.